### ISSUE TITLE
We shouldn't be putting aria-* attributes on custom elements

### ISSUE THREAD

**Issue by svanherk** [BrightspaceUI/core#1954]
[`axe-core` version `4.3.4`](https://github.com/dequelabs/axe-core/releases/tag/v4.3.4) caused a number of our tests to start breaking because we place `aria-expanded` on things like `d2l-button`, `d2l-subtle-button`, etc.  We then pass down the value into the `aria-expanded` attribute on the actual native `button`. Even though this _works_ from a screenreader POV, we really should be using something like just `expanded` here.  This may be an issue for other `aria-*` attributes as well.

The change was reverted in [`4.3.5`](https://github.com/dequelabs/axe-core/releases/tag/v4.3.5) after discussion in this issue: https://github.com/dequelabs/axe-core/issues/3241
It's still unknown whether this will be added back in version `4.4.0`, or if it will be added as a warning.  But we should go through and fix these when we can to avoid lots of warnings (in both `core` tests and everywhere that uses `d2l-button` tests) or having to lock our `axe-core` version everywhere.

**Comment by svanherk** [comment]
[Update](https://github.com/dequelabs/axe-core/issues/3241#issuecomment-1009229447) - `axe-core` `v4.4.0` (releasing at the end of this month) will add back the violation for non-global ARIA attributes on elements with a role. Custom elements will report as Needs Review rather than a failure.

So this shouldn't fail our builds (unless we have violations on native elements), and we should get a log to help us track down places we need to fix.

**Comment by KaiPrince** [comment]
Does this include `role`?

**Comment by svanherk** [comment]
I _believe_ `role` is fine and sometimes necessary, and we've definitely leveraged that in a few places. [`role`](https://www.w3.org/TR/html-aria/#el-autonomous-custom-element) and any [global aria properties](https://www.w3.org/TR/wai-aria-1.1/#global_states) won't fail the tests, but I still think we'd want to avoid the global aria-* properties if all we're doing is passing them down to the web component.  Like if we want to put an optional `aria-live` on an internal element, we should have the custom element's property just be `live` (or whatever).
